spec/spec_helper.rb

```diff
@@ -1,41 +1,41 @@
 require 'rails'
 require 'sunspot_mongo'
-require 'support/rails'
 
-ENV['MONGOID_VERSION'] = '4' unless ENV['MONGOID_VERSION'] || ENV['MONGO_MAPPER_VERSION']
+ROOT_PATH = Pathname(__FILE__).join('../..')
+Rails.define_singleton_method(:root) { ROOT_PATH }
 
 if ENV['MONGOID_VERSION']
   require 'mongoid'
   require 'support/models/mongoid'
+
+  Mongoid.connect_to('sunspot_mongo_test')
+
+  case ENV['MONGOID_VERSION']
+  when /^3|4/
+    Mongoid.default_session.drop
+  when /^5/
+    Mongo::Logger.level = Logger::INFO
+    Mongoid.default_client.database.drop
+  end
+
 elsif ENV['MONGO_MAPPER_VERSION']
   require 'mongo_mapper'
   require 'support/models/mongo_mapper'
+
+  connection = Mongo::Connection.new
+  database = connection.db('sunspot_mongo_test')
+  database.collections.reject { |col| col.name =~ /^system/ }.each(&:drop)
+  MongoMapper.connection = connection
+  MongoMapper.database = 'sunspot_mongo_test'
 end
 
 # Load shared examples
 require 'shared_examples'
 
-def setup_servers_and_connections
-  FileUtils.mkdir_p '/tmp/sunspot_mongo_test/'
-
-  @solr_pid = fork { `sunspot-solr start --log-file=solr.log --data-directory=data --log-level=INFO --port=8900` }
-  sleep 5
-
-  if ENV['MONGOID_VERSION']
-    Mongoid.configure do |config|
-      config.connect_to('sunspot_mongo_test')
-    end
-  elsif ENV['MONGO_MAPPER_VERSION']
-    connection = Mongo::Connection.new
-    database = connection.db('sunspot_mongo_test')
-    database.collections.reject { |col| col.name =~ /^system/ }.each(&:drop)
-    MongoMapper.connection = connection
-    MongoMapper.database   = 'sunspot_mongo_test'
-  end
+@solr_pid = fork { `sunspot-solr start --log-file=solr.log --data-directory=data --log-level=INFO --port=8900` }
+sleep 5
```

Comment:
@dblock Hopefully https://github.com/derekharmel/sunspot_mongo/pull/17 will fix this